execution/adapter/ecs_adapter.go

```diff
@@ -201,17 +201,51 @@ func (a *ecsAdapter) AdaptRun(definition state.Definition, run state.Run) ecs.Ru
 		overrides.TaskRoleArn = a.taskRoleArn
 	}
 
+	placementConstraints := []*ecs.PlacementConstraint{a.gpuPlacementConstraints(definition, run)}
```

Comment:
How does AWS behave when `gpuPlacementConstraints` returns `nil` for most things, meaning the first entry in the slice is a nil pointer?

Reply:
good catch updated